### PR TITLE
Review for sc UI

### PR DESCRIPTION
**Pull request by WIN-LIN**
# Description
- Rename route from `/authorization/transaction` to `/review` to align with Figma design
- Implement conditional UI rendering based on `tx.data` to differentiate contract interactions

# Screenshot
截圖 2024-07-12 下午6 28 38

### CODE REVIEW

**Comment by oneleo** [comment]
Looks good to me!

File: app/page/review/transaction.tsx [hunk 1]
```diff
@@ -148,6 +101,8 @@ function UserOperationConfirmation(props: {
     markERC4337TransactionRejected
   } = useAction()
 
+  const isContract = tx.data !== "0x"
```
Comment (pilagod) [inline app/page/review/transaction.tsx:104]: Also `||` check `tx.to` has code or not with `provider.getCode`.

File: app/page/review/transaction.tsx [hunk 2]
```diff
@@ -307,25 +262,32 @@ function UserOperationConfirmation(props: {
...
           </div>
         </div>
-        <h2 className="py-[12px]">To</h2>
+        <h2 className="py-[12px]">{isContract ? "to interact with" : "To"}</h2>
         <div className="flex gap-[12px] items-center">
-          <Wallet />
+          {isContract ? <Contract /> : <Wallet />}
           <div className="py-[16px] w-[322px]">
-            <h3 className="break-words">{props.tx.to}</h3>
+            {isContract && <h3>Dapp Contract address</h3>}
```
Comment (pilagod) [inline app/page/review/transaction.tsx:287]:
```suggestion
            {isContract && <h3>Contract address</h3>}
```

Comment (WIN-LIN) [reply app/page/review/transaction.tsx]: [Remove Dapp text](https://github.com/pilagod/waallet-extension/pull/159/commits/4131a5bd22d8694aaacc1199b94d6e6b9a7ba0de)

File: app/page/review/transaction.tsx (hunk 1 above)
Comment (WIN-LIN) [reply app/page/review/transaction.tsx]: [Also check if tx.to has code](https://github.com/pilagod/waallet-extension/pull/159/commits/94087cce23574dfe89adec998e45bdddb5c91346)

**Comment by WIN-LIN** [comment]
Currently, we treat send tokens to **AA wallet** as interacting with the contract. Therefore, it will show "Interact with contract" instead of "Send". ([slack thread](https://imtoken.slack.com/archives/C07B4GMKBQT/p1720777015098189))

[**Future work**] For better user experiences, we need find a way to tell if a contract is AA; otherwise, users may be confused about why they're sending to a contract, not an account.

**Comment by pilagod** [comment]
I thought our conclusion this morning is to use only data to check if it is a contract interaction 🤔 If it is the case, send to AA wallet would have `0x` data and it would show `Send`.

**Review by pilagod — APPROVED** [review]

**Pull request merged**